src/ThreeEditor/js/Viewport.js

```diff
@@ -482,7 +484,7 @@ export function Viewport(
 
         transformControls.detach();
 
-        if (object !== null && object !== scene && object !== camera) {
+        if (object !== null && object !== scene && object !== camera && object !== zones && !object?.unionOperations) {
```

Comment:
This seems pretty complex condition, can you add some comment ?

Reply:
Moved to separate function